sdks/python/container/boot.go

```diff
@@ -172,12 +186,12 @@ func main() {
 		}
 	}
 
-	workerPoolId := os.Getenv(workerPoolIdEnv)
 	if workerPoolId != "" {
-		multiProcessExactlyOnce(materializeArtifactsFunc, "beam.install.complete."+workerPoolId)
+		defer multiProcessExactlyOnce(materializeArtifactsFunc, "beam.install.complete." + workerPoolId)()
```

Comment:
why do we want to defer this call now?

Reply:
We do not defer this call, we defer its return value, which is an anonymous cleanup function to delete the venv.